promotion/src/main/java/org/cryptimeleon/incentive/promotion/PromotionReward.java

```diff
@@ -0,0 +1,12 @@
+package org.cryptimeleon.incentive.promotion;
+
+import lombok.Value;
+
+/**
+ * Represents what a user can gain from spending a certain amount of points.
+ */
+@Value
+public class PromotionReward {
```

Comment:
This probably needs to be serializable or something. Somehow at least, the client needs to tell the provider which reward it wants to accept.

Reply:
Deferred to next branch, I decided to divide the feature in two since this branch became huge